main.go

```diff
@@ -98,14 +97,17 @@ func main() {
 	}
 
 	watchNamespace, ok := os.LookupEnv("WATCH_NAMESPACE")
-	if ok {
+	// We can't setup managerOptions.Cache.DefaultNamespaces = map[cache.AllNamespaces]cache.Config{} due to
+	// https://github.com/kubernetes-sigs/controller-runtime/issues/2628
+	if ok && watchNamespace != "nil" {
 		managerOptions.Cache.DefaultNamespaces = map[string]cache.Config{}
 		if strings.Contains(watchNamespace, ",") {
 			for _, namespace := range strings.Split(watchNamespace, ",") {
 				managerOptions.Cache.DefaultNamespaces[namespace] = cache.Config{}
 			}
 		} else {
 			managerOptions.Cache.DefaultNamespaces[watchNamespace] = cache.Config{}
+			managerOptions.LeaderElectionNamespace = watchNamespace
```

Comment:
Ok, this allows to isolate leader-election and watched/managed namespace.

This are still may be conflict in webhooks.
I guess, kube-api will call validating/mutating hooks for all installed operators.

We could solve that in separate PR.